Mathlib/Topology/EMetricSpace/Lipschitz.lean

```diff
@@ -47,59 +47,90 @@ open Filter Function Set Topology NNReal ENNReal Bornology
-    {f : α → β} (hf : LipschitzOnWith K f t) (h : s ⊆ t) : LipschitzOnWith K f s :=
+theorem LipschitzOnWith.mono (hf : LipschitzOnWith K f t) (h : s ⊆ t) : LipschitzOnWith K f s :=
   fun _x x_in _y y_in => hf (h x_in) (h y_in)
 #align lipschitz_on_with.mono LipschitzOnWith.mono
 
+lemma LocallyLipschitzOn.mono (hf : LocallyLipschitzOn t f) (h : s ⊆ t) : LocallyLipschitzOn s f :=
+  fun x hx ↦ by obtain ⟨K, u, hu, hfu⟩ := hf (h hx); exact ⟨K, u, nhdsWithin_mono _ h hu, hfu⟩
+
 /-- `f` is Lipschitz iff it is Lipschitz on the entire space. -/
-@[simp]
-theorem lipschitzOn_univ [PseudoEMetricSpace α] [PseudoEMetricSpace β] {K : ℝ≥0} {f : α → β} :
-    LipschitzOnWith K f univ ↔ LipschitzWith K f := by simp [LipschitzOnWith, LipschitzWith]
-#align lipschitz_on_univ lipschitzOn_univ
+@[simp] lemma lipschitzOnWith_univ : LipschitzOnWith K f univ ↔ LipschitzWith K f := by
+  simp [LipschitzOnWith, LipschitzWith]
+#align lipschitz_on_univ lipschitzOnWith_univ
+
+@[deprecated (since := "2024-06-20")] alias lipschitzOn_univ := lipschitzOnWith_univ
 
-theorem lipschitzOnWith_iff_restrict [PseudoEMetricSpace α] [PseudoEMetricSpace β] {K : ℝ≥0}
-    {f : α → β} {s : Set α} : LipschitzOnWith K f s ↔ LipschitzWith K (s.restrict f) := by
+@[simp] lemma locallyLipschitzOn_univ : LocallyLipschitzOn univ f ↔ LocallyLipschitz f := by
+  simp [LocallyLipschitzOn, LocallyLipschitz]
+
+theorem lipschitzOnWith_iff_restrict : LipschitzOnWith K f s ↔ LipschitzWith K (s.restrict f) := by
   simp only [LipschitzOnWith, LipschitzWith, SetCoe.forall', restrict, Subtype.edist_eq]
 #align lipschitz_on_with_iff_restrict lipschitzOnWith_iff_restrict
 
+@[simp] lemma lipschitzOnWith_restrict {t : Set s} :
+    LipschitzOnWith K (s.restrict f) t ↔ LipschitzOnWith K f (s ∩ Subtype.val '' t) := by
+  simp only [LipschitzOnWith, LipschitzWith, Subtype.forall, restrict, Subtype.edist_eq]; aesop
+
+lemma locallyLipschitzOn_iff_restrict :
+    LocallyLipschitzOn s f ↔ LocallyLipschitz (s.restrict f) := by
+  simp only [LocallyLipschitzOn, LocallyLipschitz, SetCoe.forall', restrict, Subtype.edist_eq,
+    ← lipschitzOnWith_iff_restrict, lipschitzOnWith_restrict, nhds_subtype_eq_comap_nhdsWithin,
+    mem_comap]
+  congr! with x K
+  constructor
+  · rintro ⟨t, ht, hft⟩
+    exact ⟨_, ⟨t, ht, Subset.rfl⟩, hft.mono <| inter_subset_right.trans <| image_preimage_subset ..⟩
+  · rintro ⟨t, ⟨u, hu, hut⟩, hft⟩
+    exact ⟨s ∩ u, Filter.inter_mem self_mem_nhdsWithin hu,
+      hft.mono fun x hx ↦ ⟨hx.1, ⟨x, hx.1⟩, hut hx.2, rfl⟩⟩
+
 alias ⟨LipschitzOnWith.to_restrict, _⟩ := lipschitzOnWith_iff_restrict
+alias ⟨LocallyLipschitzOn.restrict, _⟩ := locallyLipschitzOn_iff_restrict
 #align lipschitz_on_with.to_restrict LipschitzOnWith.to_restrict
 
-theorem MapsTo.lipschitzOnWith_iff_restrict [PseudoEMetricSpace α] [PseudoEMetricSpace β] {K : ℝ≥0}
-    {f : α → β} {s : Set α} {t : Set β} (h : MapsTo f s t) :
+lemma Set.MapsTo.lipschitzOnWith_iff_restrict [PseudoEMetricSpace α] [PseudoEMetricSpace β]
```

Comment:
Thanks for catching this oversight! Does this need a deprecation also?
(I'm sympathetic to saying "no, this lemma was not used".)

Reply:
no, this lemma was not used :)

Reply:
Remove `[PseudoEMetricSpace α] [PseudoEMetricSpace β]` as they are now available as variables?